### PR TITLE
WIP: Add namespace model and its watcher

### PR DESCRIPTION
**Pull request by diemtvu**
We want to move namespace-level mTLS settings to namespace annotation. This requires pilot got inform of all namespaces (and their annotations/labels) changes. This PR introduces Istio namespace model and equip ServiceDiscovery to watch for the namespace resources.

Issue:
https://github.com/istio/istio/issues/19083

### CODE REVIEW

**Comment by diemtvu** [comment]
@costinm  I'm still working on non-k8s details and tests but want to check if this is on the right track. Can you take a look? Thanks.

**Comment by diemtvu** [comment]
Hi @costinm , do you have any early feedback on this?

**Comment by rshriram** [comment]
We need to resolve the annotation vs sidecar crd issue first before going down this path. Let us please not go down this annotation path for core networking api without proper thought. There is no coming back

File: pilot/pkg/config/coredatamodel/discovery.go [hunk 1]
```diff
@@ -520,3 +520,15 @@ func (d *MCPDiscovery) AppendInstanceHandler(f func(*model.ServiceInstance, mode
...
+
+// AppendNamespaceHandler Not Supported
+func (d *MCPDiscovery) AppendNamespaceHandler(f func(*model.Namespace, model.Event)) error {
+	log.Warnf("AppendNamespaceHandler %s", errUnsupported)
+	return nil
+}
+
+// Namespaces Not supported
+func (d *MCPDiscovery) Namespaces() ([]*model.Namespace, error) {
+	log.Warnf("Namespaces %s", errUnsupported)
```
Comment (howardjohn) [inline pilot/pkg/config/coredatamodel/discovery.go]: won't this log every time?

**Comment by diemtvu** [comment]
Not yet needed.

**Pull request closed without merging**